### PR TITLE
Properly free memory from libcurl functions

### PR DESCRIPTION
**Pull request by dfandrich**
These pointers must be freed with curl_free, not free. Also, free some
memory that leaked otherwise. The test suite now passes with valgrind.

### CODE REVIEW

File: trurl.c [hunk 1]
```diff
@@ -225,11 +225,11 @@ static void pathadd(struct option *o, const char *path)
   struct curl_slist *n;
   char *urle = curl_easy_escape(NULL, path, 0);
   if(urle) {
-    n = curl_slist_append(o->append_path, urle);
+    n = curl_slist_append(o->append_path, strdup(urle));
```
Comment (jay) [inline trurl.c:228]: this expects valid param !=NULL so I think an additional check is needed, eg
char *dupe = strdup(urle);
if(!dupe)
....

Comment (jay) [reply trurl.c:228]: actually on second thought curl_slist_append should duplicate the data so actually I'm not sure what you're doing here

Comment (bagder) [reply trurl.c:228]: Yes, `curl_slist_append()` strdups the data it self so this should not be necessary!

Comment (bagder) [reply trurl.c:228]: This causes a leak that we can prove by running:

~~~
valgrind --leak-check=full ./trurl curl.se --append path=moo

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**